### ISSUE TITLE
Static loggers for better performance

### ISSUE THREAD

**Issue by milczarekIT** [ta4j/ta4j#494]
I noticed that some loggers in ta4j are not static e.g. in class `BaseBarSeries`. It has an impact on performance and JVM memory usage.
Let's imagine that we have a software/bot which uses 1000 instances of `BaseBarSeries`. In current implementation 1000 of exactly the same loggers will be created within `BaseBarSeries` instead of 1 instance. It has an impact not only of a heap usage but it also means slower `BaseBarSeries` initialization - for each instance logger is initialized during object construction.

I created PR with solving this issue: https://github.com/ta4j/ta4j/pull/493

**Comment by team172011** [comment]
That's true, I am also a fan of static loggers, it allows you logging also in static context

**Comment by nimo23** [comment]
Another alternative would be to use lazy initialization of the logger. With this, **no** logger instance is created if the class/instance does not use it: 

```
// lazy initialization of static logger
static class LogHolder {
    static final Logger logger = Logger.getLogger(LogInterceptors.class);
}
private static Logger log() {
    return LogHolder.logger;
}
```

However, I think, for your solution is absolutly sufficient.

**Comment by nimo23** [comment]
Another alternative would be to use **lazy initialization of the static logger**. With this, **no** logger instance is created if the class/instance does not use it: 

```
public class BaseBarSeries {

// lazy initialization of static logger
static class LogHolder {
    static final Logger logger = Logger.getLogger(BaseBarSeries.class);
}
private static Logger log() {
    return LogHolder.logger;
}
..
}
```

However,  your solution is absolutly sufficient.

**Comment by team172011** [comment]
But with lazy initialization once would completely hide the possiblitiy to use logging levels, because you always have to check the current logging level before calling log() or you would add unecessary log() calls.
merged #493
